app/models/mixins/purging_mixin.rb

```diff
@@ -200,24 +200,33 @@ def purge_in_batches(conditions, window, total = 0, total_limit = nil)
           query = query.limit(current_window)
         end
 
-        if respond_to?(:purge_associated_records)
-          # pull back ids - will slow performance
-          batch_ids = query.pluck(:id)
-          break if batch_ids.empty?
-
-          current_window = batch_ids.size
-        else
-          batch_ids = query
+        count = 0
+
+        # Purge each batch with their associated records
+        # in a transaction to avoid leaving orphaned references.
+        transaction do
```

Comment:
Hide whitespace to minimize the changes...
I needed to move the early breaks out of the transaction to avoid that without completing the transaction properly.

![image](https://github.com/user-attachments/assets/801cf89d-a81b-4eae-9c3f-5eb6bf0f3352)